plant-growth/src/app/components/hooks/use-translation.tsx

```diff
@@ -48,17 +49,18 @@ const Translation = (props: ITranslationProps) => {
       setClicked(false);
     });
 
+    const isActive = isAudioPlaying && clicked;
+    const isDisabled = isRunning || (isAudioPlaying && !clicked);
+
     const handlePlay = () => {
+      console.log("isAudioPlaying", isAudioPlaying);
```

Comment:
I guess this should be removed.

Reply:
Hi Piotr, in response to your question about the difference between clicked and isAudioPlaying -- isAudioPlaying is a general check that any audio is playing at all. Clicked is a check specific to the string, that it was clicked. The difference is importance for determining which class to assign to the particular string so that we can highlight the string that is clicked when isAudioPlaying is true.

This separation is also why I currently have the hook and the translation component separated -- the isAudioPlaying state is passed down to the Translation component so that the state isn't particular to each instance of the component.

Would it be helpful to rename the variables to isAnyAudioPlaying and selectedAudio and add comments to the file?

Also, if you think there is a better way to address this set of concerns, please let me know.

Reply:
Okay, it makes perfect sense :wink: Maybe I was still lacking some sleep. I think `isAnyAudioPlaying` is a good and simple idea, I bet it could help me realize this logic immediately. Thanks.

I'd also take a look at these lines:
https://github.com/concord-consortium/alaskan-simulations/blob/874fc20b9723bfda997f27dd1ef358a6636ed9fe/plant-growth/src/app/components/hooks/use-translation.tsx#L43-L50

It's not related to this PR, but I'd try to fix it while you're there.
It should be probably something like that:

```javascript
useEffect(() => {
  if (currentAudio) {
    const handlePlaying = { ... };
    const handleEnded = { ... };
    currentAudio.addEventListener("playing", handlePlaying);
    currentAudio.addEventListener("ended", handleEnded);
   // cleanup
    return () => {
       currentAudio.removeEventListener("playing", handlePlaying);
       currentAudio.removeEventListener("ended", handleEnded);
    };
  }
}, [currentAudip]);
```

Note that the current code will not cause any bugs, as both handlePlaying and handleEnded are idempotent. But once they're not, it might cause a hard-to-find bug. Generally, it's nice to stick to this pattern of useEffect + cleanup function while dealing with native event handlers.

Reply:
Ahh I see. Thank you :) I'll add these changes.

Reply:
Hi Piotr, I added the changes as you suggested and it's working well. I also noticed there was a regression where even if the readAloudMode wasn't on the audio elements were play-able and fixed that too.